.github/workflows/compile.yml

```diff
@@ -25,6 +25,7 @@ jobs:
         os:
         - macOS-13
         - macOS-14
+        - macOS-15
```

Comment:
macos-14 and macos-15 are fine for running unit test but you cannot run vms in the tests since the runner is a vm, and nested virtualization is not supported on the GitHub runners.